app/services/katello/pulp/repository.rb

```diff
@@ -0,0 +1,22 @@
+module Katello
+  module Pulp
+    class Repository < ::Actions::Pulp::Abstract
+      attr_accessor :repo
+
+      def initialize(repo)
+        @repo = repo
+      end
+
+      def sync(overrides = {})
+        sync_options = {}
+        sync_options[:max_speed] = SETTINGS.dig(:katello, :pulp, :sync_KBlimit)
+        sync_options[:num_threads] = SETTINGS.dig(:katello, :pulp, :sync_threads)
+        sync_options[:feed] = overrides[:source_url] if overrides[:source_url]
+        sync_options[:validate] = !SETTINGS.dig(:katello, :pulp, :skip_checksum_validation)
+        sync_options.merge!(overrides[:options]) if overrides[:options]
```

Comment:
I don't see how 'options' would ever get passed in here?

Reply:
Updated to send input[:options] from the action..